nuxeo/templates/_helpers.tpl

```diff
@@ -9,12 +9,119 @@ Expand the name of the chart.
 {{/*
 Create a default fully qualified app name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
 */}}
 {{- define "nuxeo.fullname" -}}
 {{- if .Values.fullnameOverride -}}
-{{- printf .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
 {{- else -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
 {{- end -}}
+{{- end -}}
+
+{{/*
+Compile all warnings into a single message, and call fail.
+*/}}
+{{- define "nuxeo.validateValues" -}}
+{{- $messages := list -}}
+{{- $messages := append $messages (include "nuxeo.validateValues.clustering" .) -}}
+{{- $messages := append $messages (include "nuxeo.validateValues.binaryStorage" .) -}}
+{{- $messages := append $messages (include "nuxeo.validateValues.database" .) -}}
+{{- $messages := append $messages (include "nuxeo.validateValues.kafkaRedis" .) -}}
+{{- $messages := without $messages "" -}}
+{{- $message := join "\n" $messages -}}
+
+{{- if $message -}}
+{{-   printf "\n\nVALUES VALIDATION:\n%s" $message | fail -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return true if a cloud provider is enabled for binary storage.
+*/}}
+{{- define "nuxeo.cloudProvider.enabled" -}}
+{{- if or .Values.googleCloudStorage.enabled .Values.amazonS3.enabled -}}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Validate clustering configuration: must enable a cloud provider for binary storage if more than 1 replica.
```

Comment:
What do you think about adding a check on Kafka/Redis presence in such case?

Reply:
:+1:, done